lib/inv_pcg_quda.cpp

```diff
@@ -30,139 +31,133 @@ namespace quda {
-    cudaColorSpinorField* rPre = NULL;
-    cudaColorSpinorField* minvr = NULL;
-    cudaColorSpinorField* minvrSloppy = NULL;
-    cudaColorSpinorField* p = NULL;
+    int k = 0;
+    int rUpdate = 0;
 
+    cudaColorSpinorField *minvrPre = NULL;
+    cudaColorSpinorField *rPre = NULL;
+    cudaColorSpinorField *minvr = NULL;
+    cudaColorSpinorField *minvrSloppy = NULL;
+    cudaColorSpinorField *p = NULL;
 
     ColorSpinorParam csParam(b);
     cudaColorSpinorField r(b);
-    if(K) minvr = new cudaColorSpinorField(b);
+    if (K) minvr = new cudaColorSpinorField(b);
     csParam.create = QUDA_ZERO_FIELD_CREATE;
-    cudaColorSpinorField y(b,csParam);
+    cudaColorSpinorField y(b, csParam);
 
     mat(r, x, y); // => r = A*x;
-    double r2 = xmyNorm(b,r);
+    double r2 = xmyNorm(b, r);
 
     csParam.setPrecision(param.precision_sloppy);
-    cudaColorSpinorField tmpSloppy(x,csParam);
-    cudaColorSpinorField Ap(x,csParam);
+    cudaColorSpinorField tmpSloppy(x, csParam);
+    cudaColorSpinorField Ap(x, csParam);
 
     cudaColorSpinorField *r_sloppy;
-    if(param.precision_sloppy == x.Precision())
-    {
+    if (param.precision_sloppy == x.Precision()) {
       r_sloppy = &r;
       minvrSloppy = minvr;
-    }else{
+    } else {
       csParam.create = QUDA_COPY_FIELD_CREATE;
-      r_sloppy = new cudaColorSpinorField(r,csParam);
-      if(K) minvrSloppy = new cudaColorSpinorField(*minvr,csParam);
+      r_sloppy = new cudaColorSpinorField(r, csParam);
+      if (K) minvrSloppy = new cudaColorSpinorField(*minvr, csParam);
     }
-  
 
     cudaColorSpinorField *x_sloppy;
-    if(param.precision_sloppy == x.Precision() ||
-        !param.use_sloppy_partial_accumulator) {
+    if (param.precision_sloppy == x.Precision() || !param.use_sloppy_partial_accumulator) {
       csParam.create = QUDA_REFERENCE_FIELD_CREATE;
-      x_sloppy = &static_cast<cudaColorSpinorField&>(x);
-    }else{
+      x_sloppy = &static_cast<cudaColorSpinorField &>(x);
+    } else {
       csParam.create = QUDA_COPY_FIELD_CREATE;
-      x_sloppy = new cudaColorSpinorField(x,csParam);
+      x_sloppy = new cudaColorSpinorField(x, csParam);
     }
 
-
     cudaColorSpinorField &xSloppy = *x_sloppy;
     cudaColorSpinorField &rSloppy = *r_sloppy;
 
-    if(&x != &xSloppy){
+    if (&x != &xSloppy) {
       copy(y, x); // copy x to y
       zero(xSloppy);
-    }else{
+    } else {
       zero(y); // no reliable updates // NB: check this
     }
 
     const bool use_heavy_quark_res = (param.residual_type & QUDA_HEAVY_QUARK_RESIDUAL) ? true : false;
 
-    if(K){
+    if (K) {
       csParam.create = QUDA_COPY_FIELD_CREATE;
       csParam.setPrecision(param.precision_precondition);
-      rPre = new cudaColorSpinorField(rSloppy,csParam);
-      // Create minvrPre 
+      rPre = new cudaColorSpinorField(rSloppy, csParam);
+      // Create minvrPre
       minvrPre = new cudaColorSpinorField(*rPre);
       commGlobalReductionSet(false);
```

Comment:
@maddyscientist  should this `commGlobalReductionSet` be enabled for every possible preconditioner? Seems suspicious...

Reply:
This is years old code....yes it's suspicious and should be fixed.  Not here though.....  I'll file an issue now.